src/GuessControl.js

```diff
@@ -1,47 +1,71 @@
-      currentGuess: "",
-    };
+//     this.state = {
+//       currentGuess: "",
+//     };
 
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleInputChange = this.handleInputChange.bind(this);
-    this.onSubmitGuess = this.onSubmitGuess.bind(this);
-  }
+//     /**
+//      * These lines are required to make the methods/functions declared on this
+//      *  class have the correct `this` object when they run.
+//      */
+//     this.handleInputChange = this.handleInputChange.bind(this);
+//     this.onSubmitGuess = this.onSubmitGuess.bind(this);
+//   }
 
-  handleInputChange(event) {
-    this.setState({ currentGuess: event.target.value });
-  }
+//   handleInputChange(event) {
+//     this.setState({ currentGuess: event.target.value });
+//   }
 
-  onSubmitGuess() {
-    // Since the values from an HTML input are strings by default,
-    //  convert to a number for the returned guess value
-    //  by passing in the string to the Number function.
-    // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number
-    this.props.onGuess(Number(this.state.currentGuess));
-    this.setState({ currentGuess: "" });
-  }
+//   onSubmitGuess() {
+//     // Since the values from an HTML input are strings by default,
+//     //  convert to a number for the returned guess value
+//     //  by passing in the string to the Number function.
+//     // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number
+//     this.props.onGuess(Number(this.state.currentGuess));
+//     this.setState({ currentGuess: "" });
+//   }
 
-  render() {
-    return (
-      <div>
-        <input
-          type="number"
-          value={this.state.currentGuess}
-          onChange={this.handleInputChange}
-        />
-        <Button onClick={this.onSubmitGuess}>Submit Guess</Button>
-      </div>
-    );
-  }
-}
+//   render() {
+//     return (
+//       <div>
+//         <input
+//           type="number"
+//           value={this.state.currentGuess}
+//           onChange={this.handleInputChange}
+//         />
+//         <Button onClick={this.onSubmitGuess}>Submit Guess</Button>
+//       </div>
+//     );
+//   }
+// }
 
-export default GuessControl;
+// export default GuessControlOld;
+import React, { useState } from "react";
+import Button from "./Button";
+
+export default function GuessControl(data) {
+  const [currentGuess, setCurrentGuess] = useState("")
+  
+  const handleInputChange = (event) => {
+    setCurrentGuess(event.target.value)
+  }
+  const onSubmitGuess = (event) => {
+    event.preventDefault()
+    data.onGuess(Number(currentGuess))
+  }
+  return (
+    <div>
+      <input
+        type="number"
+        value={currentGuess}
+        onChange={handleInputChange}
+      />
+      <Button onClick={onSubmitGuess}>Submit Guess</Button>
```

Comment:
it is good that you used controlled components, are you aware that HTML form elements work a bit differently from other DOM elements in React, because form elements naturally keep some internal state?

Reply:
yes some experience from building react and vanilla js projects using local storage. where values from text input will persist after page reloads.